src/main/java/net/mcreator/workspace/references/ReferencesFinder.java

```diff
@@ -0,0 +1,298 @@
+			}, (a, t) -> t.getUnmappedValue().replaceFirst("CUSTOM:", "").equals(sound.getName())))
+				elements.add(me);
+		});
+
+		return elements;
+	}
+
+	public static List<ModElement> searchStructureUsages(Workspace workspace, String structure) {
+		List<ModElement> elements = new ArrayList<>();
+
+		workspace.getModElements().parallelStream().forEach(me -> {
+			GeneratableElement ge = me.getGeneratableElement();
+			if (anyValueMatches(ge, String.class, e -> {
+				ResourceReference ref = e.getAnnotation(ResourceReference.class);
+				return ref != null && ref.value().equals("structure");
+			}, (a, t) -> t.equals(structure)))
+				elements.add(me);
+			else if (anyValueMatches(ge, String.class, e -> e.isAnnotationPresent(BlocklyXML.class),
+					(a, t) -> t.contains(">" + structure + "</field>")))
+				elements.add(me);
+		});
+
+		return elements;
+	}
+
+	public static List<ModElement> searchGlobalVariableUsages(Workspace workspace, String variableName) {
+		List<ModElement> elements = new ArrayList<>();
+
+		workspace.getModElements().parallelStream().forEach(me -> {
+			if (anyValueMatches(me.getGeneratableElement(), String.class, e -> e.isAnnotationPresent(BlocklyXML.class),
+					(a, t) -> t.contains("<field name=\"VAR\">global:" + variableName + "</field>"))) {
+				elements.add(me);
+			}
+		});
+
+		return elements;
+	}
+
+	public static List<ModElement> searchLocalizationKeyUsages(Workspace workspace, String localizationKey) {
+		List<ModElement> elements = new ArrayList<>();
+
+		workspace.getModElements().parallelStream().forEach(me -> {
+			GeneratableElement ge = me.getGeneratableElement();
+			if (ge != null && workspace.getGenerator().getElementLocalizationKeys(ge).contains(localizationKey)) {
+				elements.add(me);
+			} else if (anyValueMatches(ge, String.class, e -> e.isAnnotationPresent(BlocklyXML.class),
+					(a, t) -> t.contains(">" + localizationKey + "</field>"))) {
+				elements.add(me);
+			}
+		});
+
+		return elements;
+	}
+
+	/**
+	 * Scans the entire passed workspace and collects all mod elements considered to use certain value(s).
+	 *
+	 * @param workspace The project to check mod elements from for usages.
+	 * @param clazz     The class of values to be checked.
+	 * @param validIf   The predicate used to check if a field/method is considered valid.
+	 * @param condition The predicate defining the condition that the acquired values should pass.
+	 * @param <T>       The type of values to be checked.
+	 * @return List of mod elements contained in the provided workspace and considered to use certain value(s).
+	 */
+	@SuppressWarnings("unused") public static <T> List<ModElement> searchUsages(Workspace workspace, Class<T> clazz,
+			Predicate<AccessibleObject> validIf, BiPredicate<AccessibleObject, T> condition) {
+		return workspace.getModElements().parallelStream()
+				.filter(me -> anyValueMatches(me.getGeneratableElement(), clazz, validIf, condition)).toList();

```

Comment:
Is it thread-safe that I extract `GeneratableElement`s in scope of `.parallelStream()`?

Reply:
No, good catch (self review hehe)